test/test_query.cpp

```diff
@@ -12218,5 +12218,66 @@ TEST(Query_StringOrLongStrings)
     }
 }
 
+TEST(Query_LinksWithIndex)
```

Comment:
I think we need unittests demonstrating correct operation across table changes and sync_if_needed()

Reply:
Test case modified.